lib/odometry-ng/include/odometer/OdometerDifferential.hpp

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include <cstdint>
+
+#include "encoder/EncoderInterface.hpp"
+#include "OdometerInterface.hpp"
+#include "OdometerDifferentialParameters.hpp"
+
+namespace cogip {
+
+namespace odometer {
+
+class OdometerDifferential: public OdometerInterface {
+
+public:
+    /// @brief Construct a new differential odometer object
+    /// @param parameters Odometer parameters
+    explicit OdometerDifferential(OdometerDifferentialParameters &parameters) : parameters_(parameters), left_encoder_(NULL), right_encoder_(NULL) {}
+
+    /// @brief Set the default odometry position
+    ///
+    /// @note this function should be called to reset robot position and defined a new default one
+    ///
+    /// @param x X coordinate (mm)
+    /// @param y Y coordinate (mm)
+    /// @param O angle (deg)
+    void set_position(double x, double y, double O) override
+    {
+        pos_.x = x;
+        pos_.y = y;
+        pos_.O = DEG2RAD(O);
+    }
+    
+
+    /// @brief Set the default odometry position
+    ///
+    /// @note this function should be called to reset robot position and defined a new default one
+    ///
+    /// @param pos postion reference
+    void set_position(const cogip::cogip_defs::Pose &pos) override
+    {
+        pos_.x = pos.coords().x();
+        pos_.y = pos.coords().y();
+        pos_.O = DEG2RAD(pos.O());
+    }
+    
+    /// @brief Copy current position using cogip def format
+    /// @note Data units:
+    ///         - x, y: mm
+    ///         - O: deg
+    /// @param pose cogip::cogip_defs::Pose current pose
+    void copy_position(cogip::cogip_defs::Pose &pose) override
```

Comment:
call it `set_position` as its signature is different it works thanks to c/c++ function overloading

Reply:
This is more a getter that copy the current position to the input pose reference. Maybe i can rename it `get_position` ?